hello-world/Makefile

```diff
@@ -1,12 +1,19 @@
+ifneq ($(KERNELRELEASE),)
 obj-m := helloworld.o
 helloworld-objs := hello_world.rust.o
-KDIR ?= /lib/modules/$(shell uname -r)/build
+
+$(src)/target/x86_64-linux-kernel/debug/libhello_world.a: $(src)/Cargo.toml $(wildcard $(src)/src/*.c)
+	cd $(src); env -u MAKE -u MAKEFLAGS cargo build -Z build-std=core,alloc --target=x86_64-linux-kernel
```

Comment:
1. Should we support people (like me) who don't have nightly as the default and need to run `cargo +nightly`? Does shipping a `rust-toolchain` file in the root directory work (or would it have to be in every module)? Should we be setting `RUSTUP_TOOLCHAIN ?= nightly`? (cf. https://github.com/rust-lang/rustup.rs#the-toolchain-file and the next section)

2. What's the motivation for `-u MAKE -u MAKEFLAGS`? (It seems defensible, I'd just like to document precisely why ... alternatively, should we be unsetting it in our kernel-cflags-finder subprocess alone?)

Reply:
1. How were you working under the previous system?

2. It seemed to be necessary at some point. It's possible that the `env_clear()` fixed this though

Reply:
`cargo +nightly build -Z build-std=core,alloc --target x86_64-linux-kernel`
(and before that `cargo +nightly xbuild --target ~/linux-kernel-module-rust/x86_64-linux-kernel-module.json`)

Reply:
Ok, so the solution is probably to support `$(CARGO)` which you can pass `cargo +nightly` to, does that sound good?

Reply:
Oh, yeah, that works and is pretty consistent with how we handle compilers and such. Sure. Not required for now unless you want to (I can change my default toolchain or something).